src/librustc_typeck/check/coercion.rs

```diff
@@ -1184,13 +1183,18 @@ impl<'gcx, 'tcx, 'exprs, E> CoerceMany<'gcx, 'tcx, 'exprs, E>
                     (self.final_ty.unwrap_or(self.expected_ty), expression_ty)
                 };
 
+                let reason_label = if label_expression_as_expected {
+                    "found because of this statement"
```

Comment:
It doesn't look like the "found because ..." label gets exercised in the UI tests?

Reply:
I think that might be a dead code branch. Originally I needed a way to correct the label for a case where we had

```
fn foo() -> impl Debug {
    (return 1, return 2)
}
```

and had the expected/found order reversed, but after adding the overlapping check in 1224 this check is no longer needed and can be removed.